src/Path.hs

```diff
@@ -264,9 +264,12 @@ isParentOf p l =
 --
 -- @parent (parent \"\/\") = \"\/\"@
 --
-parent :: Path Abs t -> Path Abs Dir
-parent (Path fp) =
-  Path (normalizeDir (FilePath.takeDirectory (FilePath.dropTrailingPathSeparator fp)))
+parent :: MonadThrow m => Path Abs t -> m (Path Abs Dir)
+parent = parseAbsDir
+  . normalizeDir
```

Comment:
I suspect that this line is redundant. Would you mind removing it and checking if everything still seems to work as expected?